### PR TITLE
Add support for organization certificates

### PR DESCRIPTION
**Pull request by rabadashTheFool**
WE2-745

Uses Common Name in case of an organization certificate instead of first and last name.

Signed-off-by: Mihkel Kivisild mihkel.kivisild@hotmail.com

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,16 +36,42 @@ The algorithm, which performs the validation of the Web eID authentication token
...
+cp ria_public_key.gpg /usr/share/keyrings/ria-repository.gpg
+echo "deb [signed-by=/usr/share/keyrings/ria-repository.gpg] https://installer.id.ee/media/ubuntu/ $(lsb_release -cs) main" > /etc/apt/sources.list.d/ria-repository.list
+```
+2. Install the _libdigidocpp-csharp_ package:
+```sh
+apt update
+apt install -y --no-install-recommends libdigidocpp-csharp
+```
+
+#### For macOS
```
Comment (metsma) [inline README.md:65]: Mac has prebuilt binaries available https://github.com/open-eid/libdigidocpp/releases

File: src/WebEid.AspNetCore.Example/Properties/launchSettings.json [hunk 2]
```diff
@@ -8,11 +8,20 @@
...
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Production"
+      }
+    },
     "IIS Express": {
       "commandName": "IISExpress",
       "launchBrowser": true,
       "environmentVariables": {
-        "ASPNETCORE_ENVIRONMENT": "Development",
+        "ASPNETCORE_ENVIRONMENT": "Production",
```
Comment (metsma) [inline src/WebEid.AspNetCore.Example/Properties/launchSettings.json:24]: Keep the development profile as default

File: src/WebEid.AspNetCore.Example/Properties/launchSettings.json [hunk 3]
```diff
@@ -8,11 +8,20 @@
     }
   },
   "profiles": {
+    "https": {
+      "commandName": "Project",
+      "dotnetRunMessages": true,
+      "launchBrowser": true,
+      "applicationUrl": "https://localhost:44391",
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Production"
```
Comment (metsma) [inline src/WebEid.AspNetCore.Example/Properties/launchSettings.json:17]: Development profile

File: src/WebEid.AspNetCore.Example/Properties/launchSettings.json (hunk 2 above)
Comment (rabadashTheFool) [reply src/WebEid.AspNetCore.Example/Properties/launchSettings.json]: Changed it to Development

File: src/WebEid.AspNetCore.Example/Pages/Welcome.cshtml.cs [hunk 4]
```diff
@@ -35,7 +35,17 @@ private static string GetPrincipalName(ClaimsIdentity identity)
...
+
+            if (!string.IsNullOrEmpty(givenName) && !string.IsNullOrEmpty(surname))
+            {
+                return $"{givenName} {surname}";
+            }
+            else
+            {
+                return identity.Claims.Where(claim => claim.Type == ClaimTypes.Name)
+                    .Select(claim => claim.Value)
+                    .SingleOrDefault();
```
[the remark this answers is not in the document]
Comment (mrts) [reply src/WebEid.AspNetCore.Example/Pages/Welcome.cshtml.cs:47]: In Java, the method `getPrincipalNameFromCertificate()` works as follows:

1. It tries to get the given name and surname from the certificate.
2. If both are present, it returns the full name.
3. If either is missing, it falls back to the common name (CN).
4. If the CN is also missing, it throws a `CertificateEncodingException`.

So, it ensures some form of identifier is always retrieved, and if not, it throws an exception. The implementation is [here](https://github.com/web-eid/web-eid-spring-boot-example/blob/main/src/main/java/eu/webeid/example/security/WebEidAuthentication.java#L56-L67)

[the remark this answers is not in the document]
Comment (mrts) [reply src/WebEid.AspNetCore.Example/Pages/Welcome.cshtml.cs:47]: Thanks for the explanation, this behavior is good enough I guess.

File: README.md [hunk 5]
```diff
@@ -36,16 +36,42 @@ The algorithm, which performs the validation of the Web eID authentication token
...
+
+1.  Install the _libdigidocpp-3.14.4.pkg_ package or higher. The installation packages are available from  [https://github.com/open-eid/libdigidocpp/releases](https://github.com/open-eid/libdigidocpp/releases).
```
Comment (rabadashTheFool) [inline README.md:67]:
```suggestion
1.  Install the _libdigidocpp-3.17.1.pkg_ package or higher. The installation packages are available from  [https://github.com/open-eid/libdigidocpp/releases](https://github.com/open-eid/libdigidocpp/releases).
```

File: README.md [hunk 6]
```diff
@@ -36,16 +36,42 @@ The algorithm, which performs the validation of the Web eID authentication token
...
 
 ### 3. Setup the `libdigidocpp` library for signing
+
 `libdigidocpp` is a library for creating, signing and verifying digitally signed documents according to XAdES and XML-DSIG standards. It is a C++ library that has [SWIG](http://swig.org/) bindings for C#.
 
 Set up the `libdigidocpp` library as follows:
 
+#### For MS Windows
+
 1.  Install the _libdigidocpp-3.14.4.msi_ package or higher. The installation packages are available from  [https://github.com/open-eid/libdigidocpp/releases](https://github.com/open-eid/libdigidocpp/releases).
```
Comment (rabadashTheFool) [inline README.md:46]:
```suggestion
1.  Install the _libdigidocpp-3.17.1.msi_ package or higher. The installation packages are available from  [https://github.com/open-eid/libdigidocpp/releases](https://github.com/open-eid/libdigidocpp/releases).
```

File: src/WebEid.AspNetCore.Example/Properties/launchSettings.json [hunk 7]
```diff
@@ -22,6 +31,6 @@
       "launchUrl": "{Scheme}://{ServiceHost}:{ServicePort}",
       "publishAllPorts": true,
       "useSSL": true
-    }
+    },
```
Comment (rabadashTheFool) [inline src/WebEid.AspNetCore.Example/Properties/launchSettings.json:34]:
```suggestion
    }
```

File: src/WebEid.AspNetCore.Example/appsettings.json [hunk 8]
```diff
@@ -9,3 +9,4 @@
     },
     "AllowedHosts": "*"
 }
+
```
Comment (rabadashTheFool) [inline src/WebEid.AspNetCore.Example/appsettings.json:12]:
```suggestion
```

**Review by metsma — APPROVED** [review]

**Pull request merged**